src/libutil/canon-path.cc

```diff
@@ -139,3 +143,17 @@ std::string CanonPath::makeRelative(const CanonPath & path) const
 }
 
 }
+
+// Segregate this at the bottom to ensure the other operations do not
+// use IO.
+
+#include "file-system.hh"
```

Comment:
This is trying to be too clever. C++ is not a language where you can prevent impure / I/O functions from being in scope.

Reply:
OK putting it back, as discussed today in the meeting we agree `CanonPath::fromCwd` is going away anyways.